pkg/reconciler/pingsource/resources/receive_adapter.go

```diff
@@ -59,37 +47,30 @@ func MakeReceiveAdapter(args Args) *v1.Deployment {
 			Name:      args.AdapterName,
 		},
 		Spec: v1.DeploymentSpec{
-			Replicas: &replicas,
-			Selector: &metav1.LabelSelector{
-				MatchLabels: mtlabels,
-			},
 			Template: corev1.PodTemplateSpec{
-				ObjectMeta: metav1.ObjectMeta{
-					Labels: mtlabels,
-				},
 				Spec: corev1.PodSpec{
-					ServiceAccountName: args.ServiceAccountName,
 					Containers: []corev1.Container{
 						{
-							Name:  "dispatcher",
-							Image: args.Image,
-							Env:   makeEnv(args),
-
-							// Set low resource requests and limits.
-							// This should be configurable.
-							Resources: corev1.ResourceRequirements{
-								Requests: corev1.ResourceList{
-									corev1.ResourceCPU:    resource.MustParse("125m"),
-									corev1.ResourceMemory: resource.MustParse("64Mi"),
-								},
-								Limits: corev1.ResourceList{
-									corev1.ResourceCPU:    resource.MustParse("1000m"),
-									corev1.ResourceMemory: resource.MustParse("2048Mi"),
+							Name: "dispatcher",
```

Comment:
Since it seems like you only care about the container, maybe change MakeReceiveAdapter to return a container instead of deployment?

Reply:
well, you only care about the env so just return that?

Reply:
yeah I thought of that, somehow wanted to keep the PR small. Anyway good idea!

Reply:
done.